flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/api/TableConfigTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.api;
+
+import org.junit.Test;
+
+import java.time.Duration;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests for {@link TableConfig}.
+ */
+public class TableConfigTest {
+	@Test
+	public void testGetAndSetTableConfigOptions() throws Exception{
+		Class<?> configClass = TableConfig.class;
+		TableConfig config = new TableConfig();
+		for (TestSpec<?> spec: testSpecList){
+			configClass.getMethod("set" + spec.fieldName, spec.inputClass).invoke(config, spec.inputValue);
+			assertEquals(spec.expectedValue, configClass.getMethod("get" + spec.fieldName).invoke(config));
```

Comment:
I think we should avoid use the Java reflection here, otherwise, it's hard to maintain when TableConfig is refactored again in the future. 

I think we don't have much tests here, we can have a test for each method/option.